core/options.c

```diff
@@ -485,28 +552,71 @@ parse_liststring_t(liststring_t *var, void *value) {
     (*var)[MAX_LIST_OPTION_LENGTH-1] = '\0';
 }
 
+static void
+parse_fn(enum option_type_t type, void *ptr1, void *ptr2)
```

Comment:
Not sure I understand the name: it doesn't return a function nor parse a function?  Maybe "invoke_parser_for_type()" or "parse_by_type()"?